controllers/fio_controller.go

```diff
@@ -82,6 +82,21 @@ func (r *FioReconciler) newJob(ctx context.Context, cr *perfv1alpha1.Fio, crRef
 
 	backoffLimit := int32(0)
 
+	imagePullPolicy := corev1.PullPolicy(cr.Spec.Image.PullPolicy)
+	if imagePullPolicy == "" {
+		if strings.HasSuffix(cr.Spec.Image.Name, ":latest") {
+			imagePullPolicy = corev1.PullAlways
+		} else {
+			imagePullPolicy = corev1.PullIfNotPresent
+		}
+	}
+	imagePullSecrets := []corev1.LocalObjectReference{}
+	if cr.Spec.Image.PullSecret != "" {
```

Comment:
can we try to simply assign it to the CR input? I believe that should work, without additional logic (if).

Reply:
go is a typed language. You can't assign a string to an arry of LocalObjectReference, and we shouldn't wire out the latter to the CRD

Reply:
Would be very helpful to keep this file as small as possible. 
The more logic we put in, the more unit tests need to be created.

Can we try to change the underlying type so that this mangling is not required?